src/CSE.cpp

```diff
@@ -197,6 +227,19 @@ class ComputeUseCounts : public IRGraphVisitor {
             e.accept(this);
         }
     }
+
+
+    void visit(const Call *call) {
+        bool old_protect_loads_in_scope = protect_loads_in_scope;
+        if (call->is_intrinsic(Call::address_of) ||
+            call->is_intrinsic(Call::predicated_store) ||
```

Comment:
Is this necessary? Why can't we lift regular loads out of predicated loads/stores? The load inside the address_of should be protected by the address_of logic...